fdbrpc/FlowGrpcTests.cpp

```diff
@@ -0,0 +1,190 @@
+	server.registerService(make_shared<TestEchoServiceImpl>());
+	Future<Void> _ = server.run();
+
+	shared_ptr<asio::thread_pool> pool = make_shared<asio::thread_pool>(4);
+	AsyncGrpcClient<TestEchoService> client(addr.toString(), pool);
+
+	try {
+		EchoRequest request;
+		request.set_message("Ping!");
+		{
+			auto w = client.call(&TestEchoService::Stub::Echo, request);
+			// Out-of-scope
+		}
+	} catch (Error& e) {
+		ASSERT(false);
+	}
+	co_await delay(1); // So that lifetime of client stays.
+	co_return;
+}
+
+
+TEST_CASE("/fdbrpc/grpc/stream_destroy") {
+	NetworkAddress addr(NetworkAddress::parse("127.0.0.1:50501"));
+	GrpcServer server(addr);
+	server.registerService(make_shared<TestEchoServiceImpl>());
+	Future<Void> _ = server.run();
+
+	shared_ptr<asio::thread_pool> pool = make_shared<asio::thread_pool>(4);
+	AsyncGrpcClient<TestEchoService> client(addr.toString(), pool);
+
+	int count = 0;
+	try {
+		EchoRequest request;
+		request.set_message("Ping!");
+		{
+			auto stream = client.call(&TestEchoService::Stub::EchoRecv10, request);
+			auto response = co_await stream;
+			ASSERT_EQ(response.message(), "Echo: Ping!");
+		}
+		//TODO: Test if server cancels.
+	} catch (Error& e) {
+		if (e.code() == error_code_end_of_stream) {
+			ASSERT_EQ(count, 10); // Should send 10 reponses.
+			co_return;
+		}
+		ASSERT(false);
+	}
+	co_return;
+}
+
+// T_EST_CASE("/fdbrpc/grpc/basic_stream_client") {
+// 	NetworkAddress addr(NetworkAddress::parse("127.0.0.1:50502"));
+// 	GrpcServer server(addr);
+// 	server.registerService(make_shared<TestEchoServiceImpl>());
+// 	Future<Void> _ = server.run();
+
+// 	shared_ptr<asio::thread_pool> pool = make_shared<asio::thread_pool>(4);
+// 	AsyncGrpcClient<TestEchoService> client(addr.toString(), pool);
+
+// 	int count = 0;
+// 	try {
+// 		EchoRequest request;
+// 		request.set_message("Ping!");
+// 		auto stream = client.call(&TestEchoService::Stub::EchoSend10, request);
+// 		loop {
+// 			auto response = co_await stream;
+// 		    ASSERT_EQ(response.message(), "Echo: Ping!");
+// 			count += 1;
+// 		}
+// 	} catch (Error& e) {
+// 		if (e.code() == error_code_end_of_stream) {
+// 			ASSERT_EQ(count, 10); // Should send 10 reponses.
+// 			co_return;
+// 		}
+// 		ASSERT(false);
+// 	}
+// 	co_return;
+// }
+
+TEST_CASE("/fdbrpc/grpc/file_transfer") {
+	// -- Server --
+	std::string server_address("127.0.0.1:50051");
+	FileTransferServiceImpl service;
+
+	grpc::ServerBuilder builder;
+	builder.AddListeningPort(server_address, grpc::InsecureServerCredentials());
+	builder.RegisterService(&service);
+
+	std::unique_ptr<grpc::Server> server(builder.BuildAndStart());
+	std::cout << "Server listening on " << server_address << std::endl;
+
+	// -- Client --
+	auto channel = grpc::CreateChannel(server_address, grpc::InsecureChannelCredentials());
+	auto client = FileTransferClient(channel);
+	auto start = std::chrono::high_resolution_clock::now();
+	client.DownloadFile("/root/ftexample.bin", "example.bin");
```

Comment:
this is not portable. Better create file somewhere and download into a temporary directory. Remember to clean up at the end of the test.